expression/distsql_builtin.go

```diff
@@ -1132,11 +1142,6 @@ func PBToExpr(expr *tipb.Expr, tps []*types.FieldType, sc *stmtctx.StatementCont
 		return nil, err
 	}
 
-	// recover collation information
-	if collate.NewCollationEnabled() {
```

Comment:
I don't get the point here, why we need to change the code here? What caused the issue?

Reply:
code here is fine. but it seems it is duplicate and circular reference, we had already set `Charset` and `collation` in `getSignatureByPB`.

Reply:
It seems that we can remove them?

Reply:
@wjhuang2016 I think it is, we had already set `charset` and `collation` correct in `getSignatureByPB` no need to set again